### ISSUE TITLE
Chart shows duplicate projects.

### ISSUE THREAD

**Issue by rickyv3** [mendix/ChartJS#82]
Hello,

I show duplicate values within our chart, although when running the debugger everything appears to be correct. I have created a test project that I will send you via email shortly - to recreate, simply click "Create 100 Projects" and you will see duplicate projects within the chart canvas (although all looks good when running debugger). If you then either click a project from the list below, or click the Home button (left nav) the chart will reset and appear correctly.

We are able to recreate this issue in our other project for both bar and line charts.

Thank you, Jelte.

Best,
Rick

![badchartscreenshot](https://cloud.githubusercontent.com/assets/13573078/16011015/5ebb9f66-3149-11e6-9525-64012346eabb.png)

**Comment by rickyv3** [comment]
We were able to fix the duplicate project / data set issue within our actual app by removing the use of group boxes (yes!). However, the test project I submitted does not utilize group boxes so this issue may still be valid.

Secondly, we have another issue that may or may not be related to this: our charts will only refresh when another data set exists. That is, if i am showing a bar chart where data exists then the chart will display the data as I expected it to. This issue arises when I select another project (this is done with use of a listening data view) that may have not have a data set for that chart, in which case the chart will continue to show the data sets from the project / object that has data. The chart will refresh only when I select another project that does have the appropriate data - however, what should happen in my mind is that the chart should be refreshed either way and either show the new data set or show an empty chart / canvas.
